QUARKUS-2994.md

```diff
@@ -0,0 +1,27 @@
+# QUARKUS-2994 JDK 21 Support
+
+This document is a high level test plan for certification of Red Hat build of Quarkus on JDK 21.
+
+## Scope of the testing
+
+### Impact on test suites and testing automation
+* Temurin JVM 21 and OpenJDK 21 need to be added as JDKs to Quarkus QE Jenkins
+* Temurin JVM 21 and OpenJDK 21 have to be added as JVMs to Polarion
+* Bare metal jobs in extended platform testing need to be extended to run with both Temurin 21 and OpenJDK 21.
```

Comment:
Openshift jobs: I made some changes into rhel8_openshift_ts_jvm_ocp.groovy, will add it here as well.

S2I: there is `registry.access.redhat.com/ubi8/openjdk-21:latest` so we can do this testing easily. What about adding this testing  to `rhbq-3.2-rhel8-openshift-ts-jvm-ocp` job for runs on jdk 21?

All windows jobs in extended platform are marked with "jdk17". I am not sure, whether this is intentional. Do we want this replaced with jdk 21?

Upstream coverage: announcement says, that java 21 is " fully tested in our CI" (https://quarkus.io/blog/quarkus-3-5-0-released/)

Our CI: As far as I understand, we do not plan this ATM. Our CI is based[1] on baseline tests in upstream, and they use java 17[2]

[1] https://github.com/quarkus-qe/quarkus-test-suite/pull/1556
[2] https://github.com/quarkusio/quarkus/blob/main/.github/matrix-jvm-tests.json

Reply:
About Windows - how do you plan to ensure support for JDK21 on Windows, then? I'd assume that JDK21 will be supported on Windows as well as it's a supported platform now. Same note goes for ARM, by the way. I'd presume that all supported OS/archs will have JDK21 support :]

About CI - we actually already made changes to CI to run JDK21, and those should definitely be mentioned in test plan, see https://github.com/quarkus-qe/quarkus-test-suite/commit/99c2691e53136be89700c089070f40af1ab79770 - we definitely need upstream coverage to ensure supportability of JDK21 anyway.

There's another thing about our periodic CI - our Jenkins jobs used to run on both JDK11 and 17 before 11 was deprecated, see https://gitlab.cee.redhat.com/quarkus-qe/jenkins-jobs/-/commit/683d84db1fb7187e975aa7ba0b2f9a0834256328. I remember that OpenShift jobs definitely ran JDK11 + JVM + stable, JDK11 + native + stable, JDK17 + JVM + EUS, JDK17 + native + EUS. I think at least one of the JVM mode ones should be running JDK21, otherwise we'll have no real way to gauge regressions happening with supported JDK on upstream.

About upstream - how do they plan to support JDK21 without running it in CI? Btw, the JSON you linked mentions JDK21.

Reply:
Windows: I will repeat my question from the previous message "All windows jobs in extended platform are marked with "jdk17". I am not sure, whether this is intentional. Do we want this replaced with jdk 21?". If I get answer to it, I will be able to add windows support to the plan.

GH CI: I will mention current test support in the plan.

Jenkins CI: It seems, that adding Java21 + JVM + Stable is the most optimal course of action. Sounds good?

Upstream: there was a misunderstanding, I presume. Upstream team do run CI with Java 21 and that is why they support it.

Quickstarts: Focus of this effort is extended platform. Extended platform doesn't run quickstarts on RHEL right now. Do we want to add them?

Reply:
This has been addressed offline, and is ok now.